section_to_course/admin.py

```diff
@@ -0,0 +1,347 @@
+            org=org,
+            number=number,
+            run=run,
+            fields={'display_name': self.cleaned_data['new_course_name']}
+        )
+        return paste_from_template(
+            destination_course_key=course.id,
+            source_block_usage_key=cleaned_data['source_section_id'],
+            user=self.user,
+        )
+
+    class Meta:
+        """
+        Meta configuration for SectionToCourse model form.
+        """
+
+        fields = (
+            'source_course_id',
+            'source_section_id',
+            'new_course_name',
+            'new_course_org',
+            'new_course_number',
+            'new_course_run',
+        )
+        model = SectionToCourseLink
+
+
+@admin.action(description=_('Refresh section content from source.'))
+def refresh_courses(_model_admin, request, queryset):
+    """Refresh selected courses in the admin."""
+    for item in queryset:
+        paste_from_template(
+            destination_course_key=item.destination_course_id,
+            source_block_usage_key=item.source_section_id,
+            user=request.user,
+        )
+
+
+class SectionToCourseLinkAdmin(admin.ModelAdmin):
+    """
+    Admin view for section to course links.
+    """
+
+    list_display = ('name', 'source_course_id', 'source_section_id', 'destination_course_id', 'last_refresh', 'link')
+    list_filter = ('source_course_id', 'destination_course_id')
+    actions = [refresh_courses]
+
+    def name(self, obj):  # pylint: disable=no-self-use
+        """
+        Display course name.
+        """
+        dest_course = modulestore().get_course(obj.destination_course_id)
+        return str(dest_course and dest_course.display_name)
+
+    def link(self, obj):  # pylint: disable=no-self-use
+        """
+        Generate a link to the course in studio for quick access.
+        """
+        return format_html(
+            _('<a href="/course/{course_id}">edit course</a>').format(course_id=obj.destination_course_id),
+        )
+
+    def get_readonly_fields(self, request, obj=None):
+        """
+        Make fields editable based on whether we're creating a new link.
+        """
+        if obj is None:
+            return tuple()
+        # Note: Even though the names of these fields are not on the custom creation form,
+        # their presence as read_only fields on the admin page causes them to be included
+        # on the admin editing page. So, we only list them if we're editing an existing link.
+        # At the moment, none of the fields should be modified manually, so they're all in
+        # this list.
+        return (
+            'source_course_id',
+            'destination_course_id',
+            'source_section_id',
+            'destination_section_id',
+            'last_refresh',
+            'link',
+        )
+
+    def get_form(self, request, obj, *args, **kwargs):
+        """
+        Change form depending on whether we're creating a new mini-course or not.
```

Comment:
What does "creating a new mini-course" mean? Would you mind making this more verbose?